### ISSUE TITLE
networking.wireguard.interfaces.<name>.peers.*.allowedIPs shouldn't create routes

### ISSUE THREAD

**Issue by mweinelt** [NixOS/nixpkgs#29001]
## Issue description

The option `networking.wireguard.interfaces.<name>.peers.*.allowedIPs` defines which source addresses a peer may deliver traffic from. NixOS currently creates routes from these allowedIPs entries.

Adding a route is problematic when the interface just provides a transfer network to other networks. Setting `AllowedIPs` to `0.0.0.0/0` will for example get you a default device route, which messes with other default routes and breaks internet connectivity.

### Steps to reproduce

1. Set `allowedIPs` to `[ "0.0.0.0/0" "::/0" ];`
2. Spot the default routes in the main table:
```
# ip -4 r | grep wg-foobar
default dev wg-foobar scope link
# ip -6 r | grep wg-foobar
default dev wg-foobar metric 1024 pref medium
```

## Technical details

* System: 18.03-pre114421.799435b7ca (Impala)
* Nix version: 1.11.4
* Nixpkgs version: 18.03pre114421.799435b7ca
* Sandboxing enabled: yes

**Comment by Mic92** [comment]
Requires an option to skip adding routes, if the user wants to use dynamic routing protocols instead.

**Comment by globin** [comment]
cc @fpletz

**Comment by dywedir** [comment]
Fixed by #29753?
